### PR TITLE
[CI] Make only one Github request to avoid getting rate-limited

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR reduce the number of GH requests made to 1 instead of 2.

### Motivation

Avoid getting rate limited.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .github/workflows/label-analysis.yml [hunk 1]
```diff
@@ -32,6 +38,8 @@ jobs:
           done
           echo "PR ${{github.event.number}} requires at least one non-triage team assignment label (label starting by 'team/')"
           exit 1
+        env:
+          LABELS: ${{ steps.pr-labels.outputs.MILESTONE }}
```
Comment (chouquette) [inline .github/workflows/label-analysis.yml:42]: Shouldn't this be `outputs.LABELS`?

File: .github/workflows/label-analysis.yml [hunk 2]
```diff
@@ -59,3 +66,5 @@ jobs:
             echo "Skip QA labels are not correct on PR ${{github.event.number}}. You must set 'qa/skip-qa' with exacly 1 justification, 'qa/done' or 'qa/no-code-change'"
             exit 1
           fi
+        env:
+          LABELS: ${{ steps.pr-labels.outputs.MILESTONE }}
```
Comment (chouquette) [inline .github/workflows/label-analysis.yml:70]: Same question here

File: .github/workflows/label-analysis.yml (hunk 1 above)
Comment (KevinFairise2) [reply .github/workflows/label-analysis.yml:42]: The way you access the labels looks weird to me. Shouldn't it be something like `jobs.fetch-labels.outputs.**`?

Comment (amenasria) [reply .github/workflows/label-analysis.yml]: @chouquette good point, forgot to fix my copy paste aha, fixed in [9718176](https://github.com/DataDog/datadog-agent/pull/22657/commits/97181767da326ad97df91a600953d2974e0bf4af) !

File: .github/workflows/label-analysis.yml (hunk 2 above)
Comment (amenasria) [reply .github/workflows/label-analysis.yml]: Thx, fixed in [9718176](https://github.com/DataDog/datadog-agent/pull/22657/commits/97181767da326ad97df91a600953d2974e0bf4af) !

File: .github/workflows/label-analysis.yml (hunk 1 above)
Comment (amenasria) [reply .github/workflows/label-analysis.yml]: @KevinFairise2 you're right, nice catch ! Fixed in [c5a1456](https://github.com/DataDog/datadog-agent/pull/22657/commits/c5a1456a88b75aea88e4d6f5406c132c9d577215).

**Review by chouquette — APPROVED** [review]
LGTM :rocket:

**Pull request merged**